### PR TITLE
39 contactform submit

### PR DESCRIPTION
**Pull request by Platura**
Adds everything necessary in order to recieve data from the contact page. Data can already be successfully sent from the frontend to the backend

### CODE REVIEW

**Review by theUpsider — COMMENTED** [review]
@MarcNor Approvest du? ^^ Nachdem die tests gefixt sind.

File: entrypoints/flask_app.py [hunk 1]
```diff
@@ -1119,6 +1120,36 @@ def get_settings_by_user_id(user_id, lms_user_id):
             return jsonify(settings), status_code
 
 
+@app.route("/user/<user_id>/<lms_user_id>/contactform",
+           methods=['POST'])
+@cross_origin(supports_credentials=True)
+def contact_form(user_id, lms_user_id):
+    method = request.method
```
Comment (MarcNor) [inline entrypoints/flask_app.py:1127]: Method is not used anywhere

File: errors.py [hunk 2]
```diff
@@ -3,6 +3,11 @@ class MissingParameterError(Exception):
     description = "Missing parameters in request."
 
 
+class ContactFormError(Exception):
+    code = 400
+    description = "The Contact Form could not be created."
```
Comment (MarcNor) [inline errors.py:8]: Would it not be something like "Contact Form could not be send"?

File: service_layer/services.py [hunk 3]
```diff
@@ -2076,6 +2076,33 @@ def get_settings_for_user(
...
+        lms_user_id,
+        report_topic,
+        report_type,
+        report_description,
+        date
+) -> dict:
+    with uow:
+        user = get_user_by_id(uow, user_id, lms_user_id)
+        if user == {}:
+            result = None
```
Comment (MarcNor) [inline service_layer/services.py:2091]: Shouldn't we raise an error here if a user is not found instead?

File: tests/e2e/test_api.py [hunk 4]
```diff
@@ -2716,6 +2756,7 @@ def test_update_le_from_moodle(
         assert key in response.keys()
 
 
+"""
```
Comment (MarcNor) [inline tests/e2e/test_api.py:2759]: Please add those tests again ;)

**Comment by theUpsider** [comment]
@Platura currenctly on deleting user -> delete his requests. Testing.

**Review by MarcNor — APPROVED** [review]
@Platura Looks good to me. Will be merged.

**Pull request merged**